pytorch_lightning/utilities/cli.py

```diff
@@ -772,12 +769,16 @@ def _prepare_subcommand_kwargs(self, subcommand: str) -> Dict[str, Any]:
         return fn_kwargs
 
 
-def _global_add_class_path(class_type: Type, init_args: Dict[str, Any] = None) -> Dict[str, Any]:
+def _global_add_class_path(
+    class_type: Type, init_args: Optional[Union[Namespace, Dict[str, Any]]] = None
+) -> Dict[str, Any]:
+    if hasattr(init_args, "as_dict"):
```

Comment:
```suggestion
    if isinstance(init_args, Namespace):
```

Reply:
this would remove the need for `type: ignore`, right?

Reply:
Right.